### PR TITLE
[4.2] Does move some db related application code to the respective model

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Moves some db functionality from the application classes to the respective models. One to purge the messages and another one to load the front templates and the back end template.

### Testing Instructions
- Log in on the back end
- Open the front

Messages must not be tested as the code is the same just in the model and is executed on login anyway, just with no messages.

### Actual result BEFORE applying this Pull Request
Templates are correctly loaded.

### Expected result AFTER applying this Pull Request
Templates are correctly loaded.

### CODE REVIEW

**Comment by BertaOctech** [comment]
I have tested this item :red_circle: unsuccessfully on 8578df2f99c690e4e9b653dee8c03c45602d9edf<br /><br />I obtained a 500 error when I applied the patch.

I will try again on a fresh install<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37615">issues.joomla.org/tracker/joomla-cms/37615</a>.</sub>

**Comment by BertaOctech** [comment]
I will add the system information when available<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37615">issues.joomla.org/tracker/joomla-cms/37615</a>.</sub>

**Comment by laoneo** [comment]
Make sure you have the most recent nightly installed before loading the patch as it needs a comooser install.

**Comment by BertaOctech** [comment]
Now I got a message:

> The patch could not be applied because it conflicts with a previously applied patch: administrator/components/com_messages/src/Model/MessagesModel.php

But it was a fresh install

Just in case I took that file from Joomla_4.2.0-alpha2-dev-Development-Full_Package.zip to make sure it was the original.

My system info is:
-------------------
=============
System Information
=============
php: Linux linuxcin 5.11.15-1-default #1 SMP Fri Apr 16 16:47:34 UTC 2021 (64fb5bf) x86_64
dbserver: mysql
dbversion: 10.4.19-MariaDB
dbcollation: utf8mb4_general_ci
dbconnectioncollation: utf8mb4_general_ci
dbconnectionencryption: 
dbconnencryptsupported: false
phpversion: 7.4.20
server: Apache/2.4.48 (Unix) OpenSSL/1.1.1k PHP/7.4.20 mod_perl/2.0.11 Perl/v5.32.1
sapi_name: apache2handler
version: Joomla! 4.2.0-alpha2-dev Development [ Kuamini ] 29-March-2022 18:41 GMT
useragent: Mozilla/5.0 (X11; Linux x86_64; rv:94.0) Gecko/20100101 Firefox/94.0

=============
PHP Settings
=============
memory_limit: 512M
upload_max_filesize: 40M
post_max_size: 40M
display_errors: true
short_open_tag: true
file_uploads: true
output_buffering: true
open_basedir: 
session.save_path: xxxxxx
session.auto_start: 0
disable_functions: 
xml: true
zlib: true
zip: true
mbstring: true
fileinfo: true
gd: true
iconv: true
intl: true
max_input_vars: 1000

=============
Configuration File
=============
offline: false
offline_message: This site is down for maintenance.<br />Please check back again soon.
display_offline_message: 1
offline_image: 
sitename: xxxxxx
editor: tinymce
captcha: 0
list_limit: 20
access: 1
debug: false
debug_lang: false
debug_lang_const: true
dbtype: mysqli
host: xxxxxx
user: xxxxxx
password: xxxxxx
db: xxxxxx
dbprefix: xxxxxx
dbencryption: 0
dbsslverifyservercert: false
dbsslkey: 
dbsslcert: 
dbsslca: 
dbsslcipher: 
force_ssl: 0
live_site: 
secret: xxxxxx
gzip: false
error_reporting: maximum
helpurl: https://help.joomla.org/proxy?keyref=Help{major}{minor}:{keyref}&lang={langcode}
offset: UTC
mailonline: true
mailer: mail
mailfrom: xxxxxx
fromname: xxxxxx
sendmail: xxxxxx
smtpauth: false
smtpuser: xxxxxx
smtppass: xxxxxx
smtphost: xxxxxx
smtpsecure: none
smtpport: 25
caching: 0
cache_handler: file
cachetime: 15
cache_platformprefix: false
MetaDesc: 
MetaAuthor: true
MetaVersion: false
robots: 
sef: true
sef_rewrite: false
sef_suffix: false
unicodeslugs: false
feed_limit: 10
feed_email: none
log_path: xxxxxx
tmp_path: xxxxxx
lifetime: 15
session_handler: database
shared_session: false
session_metadata: true
memcached_persist: true
memcached_compress: false
memcached_server_host: xxxxxx
memcached_server_port: 11211
redis_persist: true
redis_server_host: xxxxxx
redis_server_port: 6379
redis_server_db: 0
cors: false
cors_allow_origin: *
cors_allow_headers: Content-Type,X-Joomla-Token
cors_allow_methods: 
behind_loadbalancer: false
proxy_enable: false
proxy_host: 
proxy_port: 
proxy_user: xxxxxx
massmailoff: false
replyto: 
replytoname: 
MetaRights: 
sitename_pagetitles: 0
session_filesystem_path: 
session_memcached_server_host: xxxxxx
session_memcached_server_port: 11211
session_redis_persist: 1
session_redis_server_host: xxxxxx
session_redis_server_port: 6379
session_redis_server_db: 0
frontediting: 1
log_everything: 0
log_deprecated: 0

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37615">issues.joomla.org/tracker/joomla-cms/37615</a>.</sub>

**Comment by richard67** [comment]
@BertaOctech The 2nd problem comes from having another patch applied by the patchtester. This can make problems. Therefore after having tested a pull request with patchtester, first revert that patch in the patchtester before you apply another one for the next test of another pull request.

**Comment by BertaOctech** [comment]
@richard67 
Thank you for your comment. I did not have aby patch applied. I checked the interface, the database and just in case I reinstalled the site.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37615">issues.joomla.org/tracker/joomla-cms/37615</a>.</sub>

File: libraries/src/Application/AdministratorApplication.php [hunk 1]
```diff
@@ -227,44 +227,9 @@ public function getTemplate($params = false)
...
-		{
-			$query->extendWhere(
-				'OR',
-				[
-					$db->quoteName('s.client_id') . ' = 1',
-					$db->quoteName('s.id') . ' = :style',
-					$db->quoteName('e.enabled') . ' = 1',
-				]
-			)
-				->bind(':style', $admin_style, ParameterType::INTEGER);
```
Comment (Quy) [inline libraries/src/Application/AdministratorApplication.php:264]: Please delete:
https://github.com/joomla/joomla-cms/blob/8578df2f99c690e4e9b653dee8c03c45602d9edf/libraries/src/Application/AdministratorApplication.php#L24

**Comment by SumCompanyInc** [comment]
I have tested this item :white_check_mark: successfully on 3776100aa8b1e3610b8d9bdc0fa3d6f001f6acbd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37615">issues.joomla.org/tracker/joomla-cms/37615</a>.</sub>

File: administrator/components/com_messages/src/Model/MessagesModel.php [hunk 2]
```diff
@@ -166,4 +166,63 @@ protected function getListQuery()
...
+
+		// Check if auto_purge value set
+		if (\is_object($config) && $config->cfg_name === 'auto_purge')
+		{
+			$purge = $config->cfg_value;
+		}
+		else
+		{
+			// If no value set, default is 7 days
+			$purge = 7;
```
Comment (roland-d) [inline administrator/components/com_messages/src/Model/MessagesModel.php]: @laoneo We can drop the else and move the `$purge = 7` before the if statement.

**Comment by roland-d** [comment]
Thanks everybody

**Pull request merged**